integration/class-enable-mastodon-apps.php

```diff
@@ -71,4 +83,54 @@ function ( $item ) {
 
 		return $followers;
 	}
+
+	/**
+	 * Resolve external accounts for Mastodon API
+	 *
+	 * @param Enable_Mastodon_Apps\Entity\Account $user_data The user data
+	 * @param string                              $user_id   The user id
+	 *
+	 * @return Enable_Mastodon_Apps\Entity\Account The filtered Account
+	 */
+	public static function api_account_external( $user_data, $user_id ) {
+		if ( ! preg_match( '/^' . ACTIVITYPUB_USERNAME_REGEXP . '$/', $user_id ) ) {
+			return $user_data;
+		}
+
+		$uri = Webfinger_Util::resolve( $user_id );
+
+		if ( ! $uri ) {
+			return $user_data;
+		}
+
+		$acct = Webfinger_Util::uri_to_acct( $uri );
+		$data = get_remote_metadata_by_actor( $uri );
+
+		if ( ! $data ) {
+			return $user_data;
+		}
+
+		$account = new Account();
```

Comment:
To not always replace the complete we should re-use `$user_date` if from type Account. What do you think?

```suggestion
        if ( $user_data instanceOf 'Account' ) {
         		$account = $user_data;
        } else {
        		$account = new Account();
        }
```